### PR TITLE
Upgraded both guava and guava-testlib to 21.0 and made sure they shar…

### PR DESCRIPTION
**Pull request by shamsasari**
…e the same version variable

### CODE REVIEW

File: publish.properties [hunk 1]
```diff
@@ -1 +1,2 @@
 gradlePluginsVersion=0.10.2
+guavaVersion=21.0
```
Comment (Clintonio) [inline publish.properties]: The name publish.properties may be a bit of a misnomer now, but it's probably fine to rename it later.

File: buildSrc/build.gradle [hunk 2]
```diff
@@ -5,6 +5,11 @@ repositories {
...
 dependencies {
-    // Cannot use ext.guava_version here :(
-    compile "com.google.guava:guava:20.0"
+    // For sharing constants between builds
+    Properties props = new Properties()
+    file("../publish.properties").withInputStream { props.load(it) }
+
+    def guava_version = props.getProperty("guavaVersion")
+
+    compile "com.google.guava:guava:$guava_version"
```
Comment (Clintonio) [inline buildSrc/build.gradle]: I'm surprised the CanonicalizerPlugin needs Guava. Have you tested it without?

File: buildSrc/build.gradle [hunk 3]
```diff
@@ -5,6 +5,11 @@ repositories {
 }
 
 dependencies {
-    // Cannot use ext.guava_version here :(
-    compile "com.google.guava:guava:20.0"
+    // For sharing constants between builds
+    Properties props = new Properties()
```
Comment (Clintonio) [inline buildSrc/build.gradle]: Can you move this block up into the buildscript block. This is a bit of an abuse of the DSL. Use the ext. system the same as we do in the main build.gradle.

File: buildSrc/build.gradle (hunk 2 above)
Comment (shamsasari) [inline buildSrc/build.gradle]: It needs the `ByteStreams.copy` method

File: publish.properties (hunk 1 above)
Comment (shamsasari) [inline publish.properties]: I can rename it now if you have a better name

Comment (shamsasari) [inline publish.properties]: build.constants?

**Comment by shamsasari** [comment]
Back to you.

File: buildSrc/build.gradle [hunk 4]
```diff
@@ -1,10 +1,17 @@
+buildscript {
+    // For sharing constants between builds
+    Properties props = new Properties()
+    file("../publish.properties").withInputStream { props.load(it) }
+
+    ext.guava_version = props.getProperty("guavaVersion")
+}
+
```
Comment (chrisr3) [inline buildSrc/build.gradle]: Overkill? AFAIK the `buildSrc` module is for Gradle plugins that are part of the build process itself. Why do we _care_ that this Gradle plugin is using a different version of Guava than we are using in Corda?

File: test-utils/build.gradle [hunk 5]
```diff
@@ -38,7 +38,7 @@ dependencies {
     compile "junit:junit:$junit_version"
 
     // Guava: Google test library (collections test suite)
-    compile "com.google.guava:guava-testlib:19.0"
+    compile "com.google.guava:guava-testlib:$guava_version"
```
Comment (chrisr3) [inline test-utils/build.gradle]: I would have thought that this _single_ line was enough to resolve the issue really.

File: publish.properties (hunk 1 above)
Comment (Clintonio) [inline publish.properties]: How about versions.properties, since the file type is still a Java properties file.

Comment (chrisr3) [inline publish.properties]: Upgrading Corda's version of Guava from 19.0 to 21.0? Is that part of this issue?

**Comment by Clintonio** [comment]
@shamsasari Replied wrt to the renaming.

File: buildSrc/build.gradle (hunk 4 above)
Comment (shamsasari) [inline buildSrc/build.gradle]: True, but then there's no harm in having them the same.

File: test-utils/build.gradle (hunk 5 above)
Comment (shamsasari) [inline test-utils/build.gradle]: The commit message describes the issue, regardless of what the original reason was.

**Comment by shamsasari** [comment]
Renamed `publish.properties` to `constants.properties`. `version.properties` implies it contains all the version info.

**Comment by shamsasari** [comment]
Back to you.

File: test-utils/build.gradle (hunk 5 above)
Comment (chrisr3) [inline test-utils/build.gradle]: My point is that I don't believe that Gradle and Corda using different versions of Guava _is_ an issue.

Comment (Clintonio) [inline test-utils/build.gradle]: I see no issue with trying to keep version numbers consistent between the four different projects in this repo. It's not pretty but it's obvious what it's doing.

**Pull request merged**